src/Files.App/Data/Models/SidebarPinnedModel.cs

```diff
@@ -100,7 +100,7 @@ public async Task<LocationItem> CreateLocationItemFromPathAsync(string path)
 			if (res || (FilesystemResult)FolderHelpers.CheckFolderAccessWithWin32(path))
 			{
 				locationItem.IsInvalid = false;
-				if (res)
+				if (res.Result is not null)
```

Comment:
```suggestion
				if (res && res.Result is not null)
```

Reply:
It doesn't hurt to have the check but according to VS `res` isn't null here.